src/lib/work_selector/offset.ml

```diff
@@ -0,0 +1,87 @@
+open Core_kernel
+
+(** The work selection method implemented here is the following. We distinguish
+two cases:
+    1. The work pool is updated: a random [offset] is generated. The selected
+    work is the [offset]-th one of the expensive work list.
+    2. As long as the pool stays the same, the works are selected sequencially
+    from the offset. If the end of the work list is reached, the next works are
+    selected sequencially from the begining of the list. *)
```

Comment:
```suggestion
    2. As long as the pool stays the same, the works are selected sequentially
    from the offset. If the end of the work list is reached, the next works are
    selected sequentially from the beginning of the list.
```
I think re-selecting and working sequentially on a smaller work list, when n is too large, is more in the spirit of the solution. Especially given that many workers will need to reset to zero often in this case.

Reply:
In my head it makes more sense to do this; if you have `[0,1,2,3,4]` with offset `3` and query 5 jobs, you should have `[3,4,0,1,2]`.

I also think this way gives more uniform probablity than recomputing randomness each time the list gets too small.

Reply:
(I fixed typos)